messaging/src/main/java/org/axonframework/messaging/annotations/ParameterResolver.java

```diff
@@ -39,6 +41,36 @@ public interface ParameterResolver<T> {
     @Nullable
     T resolveParameterValue(@Nonnull ProcessingContext context);
 
+    /**
+     * Asynchronously resolves the parameter value from the {@code context}.
+     * <p>
+     * The default implementation wraps the synchronous {@link #resolveParameterValue(ProcessingContext)}
+     * in a completed {@link CompletableFuture}. If the synchronous method throws an exception, it is caught
+     * and returned as a failed {@link CompletableFuture}. Implementations that perform I/O or other async
+     * operations (such as loading entities from a repository) should override this method to provide true
+     * asynchronous resolution without blocking.
+     * <p>
+     * Example of true async implementation:
+     * <pre>{@code
+     * @Override
+     * public CompletableFuture<T> resolveParameterValueAsync(@Nonnull ProcessingContext context) {
+     *     return stateManager.loadEntity(entityType, id, context);
+     * }
+     * }</pre>
+     *
+     * @param context The current processing context.
+     * @return A {@link CompletableFuture} that will complete with the parameter value, or {@code null}.
+     * @since 5.0.0
+     */
+    @Nonnull
+    default CompletableFuture<T> resolveParameterValueAsync(@Nonnull ProcessingContext context) {
+        try {
+            return CompletableFuture.completedFuture(resolveParameterValue(context));
```

Comment:
Is this a useful default method?  It doesn't solve the problem at all, and people may get the impression (since it is called `async`) that it would be... async.

How about using at a minimum `supplyAsync` here?  Or just not providing a default implementation that is flawed.